helanal/helanal.py

```diff
@@ -238,20 +200,39 @@ def helix_analysis(positions, ref_axis=[0, 0, 1]):
     local_bends = np.diagonal(bend_matrix, offset=3)  # (n_res-6,)
 
     # radius of local cylinder
+    ## This appears to be calculated based on the following.
+    ##
+    ##            CA_i+2                     CA_i+1
+    ##         r  / | \                      / ^
+    ##          /   |  \Vi+1                /   \
+    ##        / θ   |   \                  / bi  \
+    ## origin-----------CA_i+1             ----->CA_i+1      
+    ##           x  | y /                  \     ^  
+    ##              |  /Vi                  \   /    
+    ##              | /                      \ /
+    ##             CA_i                     CA_i 
+    ##
+    ## If we assume atoms are ~evenly spaced around a circle, bi are ~bisectors 
+    ##   and their projections should meet at the centre of the circle
+    ##   i.e. r ~= x + y
+    ## bi = Vi-Vi+1 is the digonal of a parallelogram formed by Vi and Vi+1.
+    ##   A vector from CAi to CAi+2 is the other diangonal, so is perpendular
+    ##   to and bisects bi - i.e. x = rcos0 and y = |bi|/2
+    ## Solve for r and substitute |bi| -> adjacent_mag**0.5 to take into account 
+    ##   some difference.
     radii = (adjacent_mag**0.5) / (2*(1.0-cos_theta))  # (n_res-3,)
+
     # special case: angle b/w bisectors is 0 (should virtually never happen)
     # guesstimate radius = half bisector magnitude
     radii = np.where(cos_theta != 1, radii, (adjacent_mag**0.5)/2)
     # height of local cylinder
     heights = np.abs(mdamath.pdot(vectors[1:-1], local_axes))  # (n_res-3,)
 
-    local_helix_directions = (bisectors.T/bimags).T  # (n_res-2, 3)
-
     # get origins by subtracting radius from atom i+1
     origins = positions[1:-1].copy()  # (n_res-2, 3)
-    origins[:-1] -= (radii*local_helix_directions[:-1].T).T
+    origins[:-1] = origins[:-1] - (radii*local_helix_directions[:-1].T).T
```

Comment:
Why did you replace the compact `-=`?

Reply:
Is there a better way to write the `(radii*local_helix_directions[:-1].T).T` that avoids double transposition?

Reply:
My bad - this was a quick and dirty fix for an error I was getting when testing the code to check I understood what was going on; turns out I was using lists instead of numpy arrays so everything is fine, but I forgot to revert it before committing. 
I've replaced `radii` with `radii[:,None]` which should do the same thing without needing the double transposition, so hopefully that's nicer.